engine/schema/src/main/java/com/cloud/capacity/dao/CapacityDaoImpl.java

```diff
@@ -203,11 +205,15 @@ public class CapacityDaoImpl extends GenericDaoBase<CapacityVO, Long> implements
                 "FROM (SELECT vi.data_center_id, (CASE WHEN ISNULL(service_offering.cpu) THEN custom_cpu.value ELSE service_offering.cpu end) AS cpu, " +
                 "(CASE WHEN ISNULL(service_offering.speed) THEN custom_speed.value ELSE service_offering.speed end) AS speed, " +
                 "(CASE WHEN ISNULL(service_offering.ram_size) THEN custom_ram_size.value ELSE service_offering.ram_size end) AS ram_size " +
-                "FROM (((vm_instance vi LEFT JOIN service_offering ON(((vi.service_offering_id = service_offering.id))) " +
-                "LEFT JOIN user_vm_details custom_cpu ON(((custom_cpu.vm_id = vi.id) AND (custom_cpu.name = 'CpuNumber')))) " +
-                "LEFT JOIN user_vm_details custom_speed ON(((custom_speed.vm_id = vi.id) AND (custom_speed.name = 'CpuSpeed')))) " +
-                "LEFT JOIN user_vm_details custom_ram_size ON(((custom_ram_size.vm_id = vi.id) AND (custom_ram_size.name = 'memory')))) " +
-                "WHERE ISNULL(vi.removed) AND vi.state NOT IN ('Destroyed', 'Error', 'Expunging')";
+                "FROM vm_instance vi LEFT JOIN service_offering ON(((vi.service_offering_id = service_offering.id))) " +
```

Comment:
@soreana all your changes LGTM, I've one question - do you see this change in LIST_ALLOCATED_CAPACITY_GROUP_BY_CAPACITY_AND_ZONE causing any potential regression in the users of `findCapacityBy()` method, and should we do any kind of testing for this cc @nvazquez.

However, I do see use of `WHERE_STATE_IS_NOT_DESTRUCTIVE` which you append to the sql string, but whether it has impact on the overall sql query?

Reply:
ping @soreana cc @ravening we're waiting for your confirmation and then we can merge your PR. Any update/comment ?

Reply:
@rohityadavcloud Sorry I was away and missed this comment. What do you mean by regression?
I guess that you mean the findCapacityBy() returns different values for different cases in a new version, but it returns the same in an old version. Am I correctly understanding your comment?

Reply:
Hi @soreana yes, my concern is there are code changes in `findCapacityBy()` method and my concern is if that could cause regression to consumers of this method (outside of your use-case)?

Reply:
cc @DaanHoogland

Reply:
@soreana I don´t see any ansswer yet. Can you explain the method, how it is changed and how this impacts other uses please?

Reply:
@DaanHoogland @rohityadavcloud Sorry for late respond, I've lost track of this issue. No it doesn't change affect the `findCapacityBy()` return value. The `where` clause appended to the SQL query later in the code.

https://github.com/apache/cloudstack/pull/4438/files#diff-dfeaef8c3e4c5eb658e6d479055c7ffdc407a9e959de834b76e881e7b140faf5R474